### PR TITLE
refactor: Use operator-rs Duration

### PR DESCRIPTION
**Pull request by Techassi**
Part of https://github.com/stackabletech/operator-rs/issues/649

### CODE REVIEW

File: rust/crd/src/lib.rs [hunk 1]
```diff
@@ -295,16 +296,23 @@ pub struct SupersetConfig {
...
     /// Corresponds to ROW_LIMIT
     pub row_limit: Option<i32>,
+
     /// Maximum number of seconds a Superset request can take before timing out.
     /// This setting affects the maximum duration a query to an underlying datasource can take.
     /// If you get timeout errors before your query returns the result you may need to increase this timeout.
     /// Corresponds to SUPERSET_WEBSERVER_TIMEOUT
-    pub webserver_timeout: Option<u32>,
+    #[fragment_attrs(schemars(
+        description = "Maximum number of seconds a Superset request can take before timing out. See https://docs.stackable.tech/home/nightly/concepts/duration for more information on the duration format"
```
Comment (sbernauer) [inline rust/crd/src/lib.rs:305]: I plan to have a look at the fragment generation regarding the descriptions today (see https://github.com/stackabletech/hdfs-operator/pull/407#discussion_r1356529851)

Anyway, we should rename to something like "Maximum time period a Superset request can take before timing out"

**Review by sbernauer — CHANGES_REQUESTED** [review]
I fear this will not be possible so easy.
Something something product-config something something legacy something something
Relevant code:
https://github.com/stackabletech/superset-operator/blob/50f3937a1ad6f72bbef50adcd7a6b06b9e8ddb20/deploy/config-spec/properties.yaml#L36-L57

File: rust/crd/src/lib.rs (hunk 1 above)
Comment (Techassi) [reply rust/crd/src/lib.rs]: Yes, ideally we would use the doc comments for the description. For now, I adjusted the attribute as suggested.

**Comment by Techassi** [comment]
> I fear this will not be possible so easy. Something something product-config something something legacy something something Relevant code:
> 
> https://github.com/stackabletech/superset-operator/blob/50f3937a1ad6f72bbef50adcd7a6b06b9e8ddb20/deploy/config-spec/properties.yaml#L36-L57

I removed the property from the product config. This seems like a logical step, as we try to move away from the product config mechanism.

File: rust/crd/src/lib.rs (hunk 1 above)
Comment (Techassi) [reply rust/crd/src/lib.rs]: Fixed in 83fd303

**Comment by Techassi** [comment]
We decided to delay the CRD change until our CRD versioning is in place. This PR now only replaces `Duration` in non-CRD parts of the code.

**Review by sbernauer — APPROVED** [review]
Thanks

**Pull request merged**